scene/main/viewport.cpp

```diff
@@ -707,28 +709,44 @@ void Viewport::_process_picking() {
 		bool captured = false;
 
 		if (physics_object_capture.is_valid()) {
+			physics_object_capture = ObjectID();
+
 			CollisionObject3D *co = Object::cast_to<CollisionObject3D>(ObjectDB::get_instance(physics_object_capture));
 			if (co && camera_3d) {
 				_collision_object_3d_input_event(co, camera_3d, ev, Vector3(), Vector3(), 0);
 				captured = true;
 				if (mb.is_valid() && mb->get_button_index() == MouseButton::LEFT && !mb->is_pressed()) {
 					physics_object_capture = ObjectID();
 				}
-
-			} else {
-				physics_object_capture = ObjectID();
+			}
+			CSGShape3D *cs = Object::cast_to<CSGShape3D>(ObjectDB::get_instance(physics_object_capture));
```

Comment:
Since CollisionObject3D and CSGShape3D don't share an interface that could be used here, I had to duplicate the code.